lib/sql-string.js

```diff
@@ -193,14 +193,16 @@ SqlString.bufferToString = function(buffer, dialect) {
 };
 
 SqlString.objectToValues = function(object, timeZone) {
-  var values = [];
-  for (var key in object) {
```

Comment:
As a rule of thumb we favour maintainability over what is still unproven performance claims.
All the performance changes we make are generally heavily tested and benchmarked to see the impact if any, look at the `groupJoinData` code, we explicitely make a choice to have unmaintainable code for the sake of performance, that should only be a very rare case though :)